File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.py
def solution(n):
    x = bin(n + 1)
    n = bin(n)
    target_count = n.count('1')
    
    while x.count('1') != target_count:
        x = int(x, 2) + 1
        x = bin(x)
        
    return int(x, 2)
        
        

File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.swift
import Foundation

func solution(_ n:Int) -> Int {
    var answer = 0
    let oneCount = String(n, radix: 2).filter { $0 == "1" }.count
    
    var candidate = n + 1
    
    while true {
        let binaryCandidate = String(candidate, radix: 2)
        if binaryCandidate.filter { $0 == "1" }.count == oneCount {
            answer = candidate
            break
        }
        candidate += 1
    }
    
    return answer
}